sui_programmability/examples/nfts/sources/DiscountCoupon.move

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module NFTs::DiscountCoupon {
+    use Sui::Coin;
+    use Sui::NFT::{Self, NFT};
+    use Sui::SUI::SUI;
+    use Sui::Transfer;
+    use Sui::TxContext::{Self, TxContext};
+
+    /// Sending to wrong recipient.
+    const EWRONG_RECIPIENT: u64 = 0;
+
+    /// Percentage discount out of range.
+    const EOUT_OF_RANGE_DISCOUNT: u64 = 1;
+
+    /// Discount coupon NFT.
+    struct DiscountCoupon has store {
+        // coupon issuer
+        issuer: address,
+        // percentage discount [1-100]
+        discount: u8,
+        // expiration timestamp (UNIX time) - app specific
+        expiration: u64,
+    }
+
+    /// Simple issuer getter.
+    public fun issuer(coupon: &DiscountCoupon): address {
+        coupon.issuer
+    }
+
+    /// Mint then transfer a new `DiscountCoupon` NFT, and top up recipient with some SUI.
+    public fun mint_and_topup(
+        discount: u8,
+        expiration: u64,
+        recipient: address,
+        coin: Coin::Coin<SUI>, 
+        ctx: &mut TxContext,
+    ) {
+        assert!(discount > 0 && discount <= 100, EOUT_OF_RANGE_DISCOUNT);
+        let nft = NFT::mint(
+                DiscountCoupon { 
+                    issuer: TxContext::sender(ctx),
+                    discount: discount, 
+                    expiration: expiration,
```

Comment:
Similar to Rust, you can skip the value if the variable name is the same as the key.